testdata/templates/example_kms_key_ring_iam_member.tf

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+      version = "~> {{.Provider.version}}"
+    }
+  }
+}
+
+provider "google" {
+  {{if .Provider.credentials }}credentials = "{{.Provider.credentials}}"{{end}}
+}
+
+resource "google_kms_key_ring_iam_member" "key_ring" {
+  key_ring_id = "{{.Provider.project}}/global/keyring-example"
```

Comment:
Instead of supplying a key_ring_id, this should use a reference to a key ring resource id. For example:

```
resource "google_kms_key_ring" "example_keyring" {
  name     = "keyring-example"
  location = "global"
  project  = "{{.Provider.project}}"
}
```

This is the way we expect users to interact with these resources, and will avoid the issue of the resource not existing.

Reply:
When I try this locally, I get the following error:

```
Error: converting tfplan to CAI assets: adding resource changes to converter: adding resource create or update fetching asset: Error parsing resource ID for : Invalid KeyRing id format, expecting `{projectId}/{locationId}/{keyRingName}` or `{locationId}/{keyRingName}.`
```

It looks like this is probably because there are a couple of lines missing from the kms_key_ring `Fetch*` code, which check for an empty parent field. (They were added to the template after you'd already started your branch, I think - sorry I didn't catch this sooner.)

Here are the corresponding lines for storage bucket iam: https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/a4c23f33b71b2fc62419b0e034e39fea654bb0c3/google/storage_bucket_iam.go#L77

You'll need to make a PR to add those lines to the terraform-google-conversion code for kms key ring iam & kms crypto key iam, then update the version of terraform-google-conversion used in this PR.

Reply:
@melinath , this is taken care. please review this PR.